docs/aws/audit/elbmonitoring/rules/min_ec2_instances_available.mdx

```diff
@@ -24,6 +24,261 @@ CBP
+   ```
+   Replace `<TargetGroupArn>` with the ARN of the target group associated with the ELB. This command will return a JSON output with details of all the instances attached to the ELB.
+
+4. Count the number of instances: You can then parse the JSON output to count the number of instances attached to each ELB. If the number of instances is less than the minimum required, then it is a misconfiguration. You can use a Python script to automate this process. Here is a simple example:
+
+   ```python
+   import json
+   import subprocess
+
+   # Run the AWS CLI command to get the details of the instances attached to the ELB
+   result = subprocess.run(['aws', 'elbv2', 'describe-target-health', '--target-group-arn', '<TargetGroupArn>'], stdout=subprocess.PIPE)
+
+   # Parse the JSON output
+   data = json.loads(result.stdout)
+
+   # Count the number of instances
+   num_instances = len(data['TargetHealthDescriptions'])
+
+   # Check if the number of instances is less than the minimum required
+   if num_instances < MIN_NUM_INSTANCES:
+       print('Misconfiguration detected: Number of instances is less than the minimum required')
+   ```
+   Replace `<TargetGroupArn>` with the ARN of the target group associated with the ELB, and `MIN_NUM_INSTANCES` with the minimum number of instances required.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials and the region you want to check.
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_key',
+    region_name='your_region'
+)
+```
+
+3. Use the Boto3 client for ELB (Elastic Load Balancer) to describe the load balancers and get the instances attached to each. The following script will print the number of instances attached to each ELB.
+```python
+elb = session.client('elb')
+
+response = elb.describe_load_balancers()
+
+for load_balancer in response['LoadBalancerDescriptions']:
+    print(f"Load Balancer Name: {load_balancer['LoadBalancerName']}")
+    print(f"Number of Instances: {len(load_balancer['Instances'])}")
+```
+
+4. To check if the minimum number of EC2 instances is configured for each ELB, you can add a condition in the loop. For example, if the minimum number should be 2, you can modify the script like this:
+```python
+for load_balancer in response['LoadBalancerDescriptions']:
+    num_instances = len(load_balancer['Instances'])
+    print(f"Load Balancer Name: {load_balancer['LoadBalancerName']}")
+    print(f"Number of Instances: {num_instances}")
+    if num_instances < 2:
+        print("Misconfiguration Detected: Less than 2 instances configured for this ELB.")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>